### ISSUE TITLE
Security issues of the second threat scenario

### ISSUE THREAD

**Issue by LuisKay** [ando-khachatryan/HiDDeN#6]
Recently, I was reproducing your work.  That's really promising and inspiring. And I got some issues consequently. Emmm, whether or not the only difference among these 5 models is the random seed? Perhaps  I'd ignored some details,  but the detection accuracy of the trained model towards encoded images genereated from 6th model was over 50%, even higher... That's really confusing and it will help a lot with your comfirmation received.@ando-khachatryan

**Comment by ando-khachatryan** [comment]
Thanks. Note that this is not my work, the authors of the paper are Zhu et all, see the reference to their paper and their repo is in the readme. But I can answer most of the questions.
If by "models" you mean different Noise configurations - there is difference between them, and if the errors are not very close, then the difference is unlikely to be due to seeding.

One reason you might be getting 50% errors is that you are not setting the model into .eval() mode before running evaluation. This is done in validate_on_batch() method of Hidden object.

Please run the script  _validate-trained-models.py_ with appropriate parameters. It will run validation for all the pre-trained models (which it takes from './experiments/' but you can override it), and saves them into a csv file. You should get results similar to those I provided. 

Let me know if this helps.

**Comment by LuisKay** [comment]
Thank you very much for your reply. I'm sorry that we may not make our question clear. We are confused about the secrecy experiment in section 4.1 in the paper, especially in the second threat scenario. The paper mentioned that:
>For our model we evaluate two threat scenarios. We first train one encoder-decoder pair and evaluate ATS assuming it has access to the learned model directly. In this setting (model weights known), ATS achieves a detection accuracy of 98%, showing the susceptibility of our model to statistical analysis when the precise weights are known. In the second threat scenario, we consider the case where the attacker knows HiDDeN is being used (and furthermore, knows the precise architecture, hyperparameters and dataset) but does not have access to the trained model itself (which is not exactly reproducible without the random seed used to initialize the parameters). To test this scenario, we train five identical HiDDeN models with different seeds and provide these models to ATS for building its artificial training set. Finally, we train a sixth model with a new random initialization for evaluation. When we test ATS on this sixth model (Ours – different model), it achieves a detection accuracy of 50%, no better than random guessing. ">

We have reproduced this experiment, for the setting that model weights are known by attackers, we found ATS achieves a detection accuracy of 98%. However, for the experiment setting that model weights are unknown and the seeds are different, the result shows that the detection accuracy towards the encoded images generated from the 6th model is much higher than 50%, which does not match our expectations. In our experiment, the only difference among the 6 HiDDen models is seed. So if you have any idea where we may make mistakes, please tell us and it will be very helpful. @jirenz

**Comment by jirenz** [comment]
@LuisKay Thank you for bringing this up. The procedure that you described matches our experiments. What I might guess is that this result depends on random seed, which may have different behaviors on different platforms.

Is the detection rate on the 6th model also 98% or something like 60% or 70%? 

One thing to try is to reduce the bit rate of the encoded message. This makes the encoding less constrained, which makes it easier to learn diverse encoding.

**Comment by ando-khachatryan** [comment]
@LuisKay Sorry, I completely misunderstood the question :)) Thanks @jirenz for coming to help.

**Comment by LuisKay** [comment]
Yep, the detection accuracy in second threat scenario is over 90% .And agreed, the reduction of relative bit rate may be a feasible approach. However, except for the random seed, the datasets these 6 models based on should be different or not? That might matters.

**Comment by LuisKay** [comment]
Yep, the detection accuracy in second threat scenario is over 90%. And agreed, the reduction of relative bit rate may be a feasible approach. However, except for the random seed, the datasets these 6 models based on should be different or not? That might matters. @jirenz @ando-khachatryan

**Comment by jirenz** [comment]
The dataset should be the same. @LuisKay

**Comment by LuisKay** [comment]
Okay, thanks. Got it. @jirenz
